### PR TITLE
Add GHA as described in cloud-content-handbook

### PR DESCRIPTION
**Pull request by abikouo**
Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/1834

- Pin `ansible-lint` version to `6.21.x`
- Organizes the workflows based on https://github.com/ansible-collections/cloud-content-handbook/blob/main/CI/README.md.
- Migrate `kubernetes.core` integration tests from zuul to GHA
- Fixing linting errors

Refer https://issues.redhat.com/browse/ACA-1123

### CODE REVIEW

File: changelogs/changelog.yaml [hunk 1]
```diff
@@ -5,154 +5,154 @@ releases:
   1.1.0:
```
Comment (hakbailey) [inline changelogs/changelog.yaml:5]: Isn't this going to have the same issue as in the cloud.terraform collection?

File: .github/workflows/integration-tests-kubernetes-core.yaml [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+          fetch-depth: "0"
+
+      - name: Split all Kubernetes targets into Job
+        id: splitter
+        uses: ansible-network/github_actions/.github/actions/ansible_test_splitter@main
+        with:
+          collections_to_test: ${{ env.source_dir }}
+          total_jobs: 8
+        env:
+          ANSIBLE_TEST_ALL_THE_TARGETS: "true"
```
Comment (hakbailey) [inline .github/workflows/integration-tests-kubernetes-core.yaml:34]: Does this mean that it's always going to test all the targets?

File: .github/workflows/integration-tests-kubernetes-core.yaml [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        shell: bash
+  integration:
+      runs-on: ubuntu-latest
+      timeout-minutes: 60
+      needs:
+        - splitter
+      if: ${{ needs.splitter.outputs.test_targets != '' }}
+      env:
+        ansible_version: "milestone"
+        python_version: "3.11"
```
Comment (hakbailey) [inline .github/workflows/integration-tests-kubernetes-core.yaml:50]: Should this match the matrix python version?

Comment (abikouo) [reply .github/workflows/integration-tests-kubernetes-core.yaml:50]: this one is not used, I will remove it

File: changelogs/changelog.yaml (hunk 1 above)
Comment (abikouo) [reply changelogs/changelog.yaml:5]: Do you mean the fact that this file is generated?

File: .github/workflows/integration-tests-kubernetes-core.yaml (hunk 2 above)
Comment (abikouo) [reply .github/workflows/integration-tests-kubernetes-core.yaml:34]: Yes, this is the intent. This is a cross-collection testing, the `cloud.common` collection does not define integration tests, the goal of the integration tests here is to validate the full `kubernets.core` test suite is not impacted by changes on this collection

Comment (hakbailey) [reply .github/workflows/integration-tests-kubernetes-core.yaml:34]: Ah, got it. Thanks!

File: changelogs/changelog.yaml (hunk 1 above)
Comment (hakbailey) [reply changelogs/changelog.yaml:5]: Yeah

Comment (abikouo) [reply changelogs/changelog.yaml]: updated

File: .github/workflows/integration-tests-kubernetes-core.yaml (hunk 3 above)
Comment (abikouo) [reply .github/workflows/integration-tests-kubernetes-core.yaml]: updated

**Review by hakbailey — APPROVED** [review]
This looks good once sanity tests run (see https://github.com/ansible-network/github_actions/pull/131).

File: .github/workflows/linters.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+    branches:
+      - main
+      - stable-*
+    tags:
```
Comment (GomathiselviS) [inline .github/workflows/linters.yaml:12]: Is this needed?

**Review by GomathiselviS — APPROVED** [review]

File: .github/workflows/linters.yaml (hunk 4 above)
Comment (abikouo) [reply .github/workflows/linters.yaml:12]: it is not mandatory but adding it as an extra check

**Pull request merged**